File: Codechef/Smallest Number of Notes/Smallest Number of Notes.cpp
//TODO: A program to calculate smallest number of notes to be given

#include <iostream>


int main(void) 
{
    int t;
    cin>>t;
    for(int i = 0; i < t;i++)                   //TODO: Take user input
    {
        int n = 0;
        cin>>n;
        int count = 0;
        while (n >= 100)                        //TODO: Calculating 100 rupees note
        {
            n -= 100;
            count++;
        }
        while (n >= 50)                         //TODO: Calculate 50 rupees note
        {
            n -= 50;
            count++;
        }
        while (n >= 10)                        //TODO: Calculate 10 rupees note
        {
            n -= 10;
            count++;
        }while (n >= 5)                        //TODO: Calculate 5 rupees note
        {
            n -= 5;
            count++;
        }
        while (n >= 2)                          //TODO: Calculate 2 rupees note
        {
            n -= 2;
            count++;
        }
        while (n >= 1)                           //TODO: Calculate 1 rupee note
        {
            n -= 1;
            count++;
        }
        cout<<count<<endl;                      //TODO: Printing result
    }
}

File: Codechef/Smallest Number of Notes/Smallest Number of Notes.py
t = int(input())
for i in range(t):    # Take user input
    n = int(input())
    count = 0
    while(n >= 100):  # Calculating 100 Rupee notes
        n -= 100
        count += 1
    while(n >= 50):   # Calculating 50 Rupee notes
        n -= 50
        count += 1
    while(n >= 10):   # Calculating 10 Rupee notes
        n -= 10
        count += 1
    while(n >= 5):    # Calculating 5 Rupee notes
        n -= 5
        count += 1
    while(n >= 2):    # Calculating 2 Rupee notes
        n -= 2
        count += 1
    while(n >= 1):    # Calculating 1 Rupee notes
        n -= 1
        count += 1
    
    print(count)